lib/docker-delta.coffee

```diff
@@ -112,16 +110,27 @@ nullDisposer = ->
 
 hardlinkCopy = (srcRoot, dstRoot, linkDests) ->
 	rsyncArgs = [
-		'--timeout', "#{RSYNC_TIMEOUT}"
 		'--archive'
 		'--delete'
 	]
 	rsyncArgs.push('--link-dest', dest) for dest in linkDests
 	rsyncArgs.push(srcRoot, dstRoot)
 	rsync = spawn('rsync', rsyncArgs)
-	utils.waitPidAsync(rsync)
+	rsync.waitAsync()
 
-exports.applyDelta = (srcImage) ->
+applyBatch = (rsync, batch, timeout) ->
+	new Promise (resolve, reject) ->
+		batch.pipe(rsync.stdin).on 'unpipe', ->
+			# wait for clean exit
+			p = rsync.waitAsync()
```

Comment:
This is very nice! It won't work as-is because we don't get a chance to install a timeout before the promise from `rsync.waitAsync()` has resolved, but I did modify this part in a similar way.